------------------------------------------------------------
revno: 14159
revision-id: squid3@treenet.co.nz-20170529043741-9chwfs5onxuip52x
parent: squid3@treenet.co.nz-20170529043611-1hyb93ivtu5wrdwg
fixes bug: http://bugs.squid-cache.org/show_bug.cgi?id=3772
author: Rainer Tammer <rainer.tammer@schulergroup.com>
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Mon 2017-05-29 16:37:41 +1200
message:
  Bug 3772: message from FTP server gets mangled
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20170529043741-9chwfs5onxuip52x
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: 800db5dab62d996440fd6fccd35e9f1f34f2f0e1
# timestamp: 2017-05-29 04:40:02 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squid3@treenet.co.nz-20170529043611-\
#   1hyb93ivtu5wrdwg
# 
# Begin patch
=== modified file 'src/clients/FtpGateway.cc'
--- src/clients/FtpGateway.cc	2017-05-29 04:36:11 +0000
+++ src/clients/FtpGateway.cc	2017-05-29 04:37:41 +0000
@@ -1541,7 +1541,7 @@
         /* Reset cwd_message to only include the last message */
         ftpState->cwd_message.reset("");
         for (wordlist *w = ftpState->ctrl.message; w; w = w->next) {
-            ftpState->cwd_message.append(' ');
+            ftpState->cwd_message.append('\n');
             ftpState->cwd_message.append(w->key);
         }
         ftpState->ctrl.message = NULL;

